### PR TITLE
Remove verification on callback signup

### PR DESCRIPTION
**Pull request by plisovin**
### Trello card

[Trello-4640](https://trello.com/c/VE4kl5hZ/4640-remove-need-for-verification-on-callback-signup)

### Context

Verification step to book a callback is not necessary

### Changes proposed in this pull request

Remove verification steps from the booking callback journey

### CODE REVIEW

File: app/controllers/callbacks/steps_controller.rb [hunk 1]
```diff
@@ -50,7 +51,7 @@ def wizard_store
     end
 
     def app_store
-      session[:callbacks] ||= {}
+      session[:callbacks] ||= session[:mailinglist]&.slice("first_name", "last_name", "email", "accepted_policy_id") || {}
```
Comment (plisovin) [inline app/controllers/callbacks/steps_controller.rb:54]: Just thinking - what if a user got interrupted and comes back to the form later when the session expires. Then he won't have those values from the session and the API eventually will fail. Need to put something here to redirect back to the mailinglist form?

File: config/environments/test.rb [hunk 2]
```diff
@@ -55,4 +55,5 @@
   config.x.integration_credentials = { username: ENV["HTTP_USERNAME"], password: ENV["HTTP_PASSWORD"] }
   config.x.mailsac_api_key = ENV["MAILSAC_API_KEY"]
   config.x.dfe_analytics = true
+  config.middleware.use RackSessionAccess::Middleware
```
Comment (ethax-ross) [inline config/environments/test.rb:58]: What's this for?

File: app/controllers/callbacks/steps_controller.rb (hunk 1 above)
Comment (ethax-ross) [reply app/controllers/callbacks/steps_controller.rb:54]: Yeah that's a good catch; if the various don't exist in the ML session we would want to show them something else, maybe a session expired message or something

**Review by ethax-ross — COMMENTED** [review]
Looking along the right track to me 👍

File: app/views/mailing_list/steps/completed/shared/_book_a_callback.html.erb [hunk 3]
```diff
@@ -7,17 +7,19 @@
...
-    <%= link_to("Book a callback", callbacks_steps_path, class: "button") %>
-    <div>
-      <small>or call us now:</small>
-      <p class="telephone">0800 389 2500</p>
+    <p>
+      Call us on <a href="tel:08003892500">0800 389 2500</a>. Monday to Friday between 8:30am and 5:30pm, except on <a href="https://www.gov.uk/bank-holidays">bank holidays</a>.
+    </p>
+    </div>
+    <p>
+      Alternatively, choose a convinent time for one of our experts to give you a call. <%= link_to("Book a callback", callbacks_steps_path,) %>.
```
Comment (gemmadallmandfe) [inline app/views/mailing_list/steps/completed/shared/_book_a_callback.html.erb:15]:
```suggestion
      Alternatively, choose a convenient time for one of our experts to give you a call. <%= link_to("Book a callback", callbacks_steps_path,) %>.
```

@philspray

File: app/views/callbacks/steps/callback/_available.html.erb [hunk 4]
```diff
@@ -1,16 +1,15 @@
...
+  Enter the number you’d like us to call you on and choose the 30 minute slot that suits you best.
 </p>
 
 <p>
   All slots are in UK time.
 </p>
 
 <p>
-  Please include international dialling code where applicable.
+  If you're outside the UK, please include international dialling code.
```
Comment (gemmadallmandfe) [inline app/views/callbacks/steps/callback/_available.html.erb:12]:
```suggestion
  If you're outside the UK, please include the international dialling code.
```
Add `the` or `your` @philspray ?

File: app/views/mailing_list/steps/completed/shared/_book_a_callback.html.erb [hunk 5]
```diff
@@ -7,17 +7,19 @@
 
 <div class="mailing-list__actions">
   <% if callback_available? %>
-    <%= link_to("Book a callback", callbacks_steps_path, class: "button") %>
-    <div>
-      <small>or call us now:</small>
-      <p class="telephone">0800 389 2500</p>
+    <p>
+      Call us on <a href="tel:08003892500">0800 389 2500</a>. Monday to Friday between 8:30am and 5:30pm, except on <a href="https://www.gov.uk/bank-holidays">bank holidays</a>.
```
Comment (gemmadallmandfe) [inline app/views/mailing_list/steps/completed/shared/_book_a_callback.html.erb:11]: Not sure that the second sentence makes sense as a standalone now - not a complete sentence @philspray ?

File: app/controllers/callbacks/steps_controller.rb [hunk 6]
```diff
@@ -50,7 +51,9 @@ def wizard_store
     end
 
     def app_store
-      session[:callbacks] ||= {}
+      redirect_to session_expired_path if session[:mailinglist].blank?
```
Comment (ethax-ross) [inline app/controllers/callbacks/steps_controller.rb:54]: Not sure if it makes sense to redirect in the `app_store` method; would it be cleaner to have a `before_action` that checks if the session has expired?

**Review by ethax-ross — COMMENTED** [review]
Looking good just one minor comment. I also got a 500 error when I tested it though, I think the email field isn't passing through for some reason?

https://dfe-teacher-services.sentry.io/issues/4293201121/?project=5276949&query=is%3Aunresolved&referrer=issue-stream&stream_index=0

File: app/controllers/callbacks/steps_controller.rb (hunk 6 above)
Comment (plisovin) [reply app/controllers/callbacks/steps_controller.rb:54]: good point, thanks

**Comment by plisovin** [comment]
> Looking good just one minor comment. I also got a 500 error when I tested it though, I think the email field isn't passing through for some reason?
> 
> https://dfe-teacher-services.sentry.io/issues/4293201121/?project=5276949&query=is%3Aunresolved&referrer=issue-stream&stream_index=0

Looks like I've managed to replicate it - if you completed the journey and then go back and create a callback again the store is empty at this point and the email doesn't exists there anymore. So no need to send the request to the API

File: app/controllers/callbacks/steps_controller.rb [hunk 7]
```diff
@@ -50,7 +52,11 @@ def wizard_store
     end
 
     def app_store
-      session[:callbacks] ||= {}
+      session[:callbacks] = if session[:callbacks].nil?
```
Comment (ethax-ross) [inline app/controllers/callbacks/steps_controller.rb:55]: Could this be simplified?

```
session[:callbacks] ||= {}
session[:callbacks].merge(session[:mailinglist]&.slice("first_name", "last_name", "email", "accepted_policy_id"))
```

Comment (plisovin) [reply app/controllers/callbacks/steps_controller.rb:55]: it will do actually with ```merge!```, thanks!

**Review by ethax-ross — APPROVED** [review]
LGTM

**Comment by gemmadallmandfe** [comment]
Closing this PR - we have decided not to go ahead with this (see Trello card)

**Pull request closed without merging**